components/eam/src/physics/crm/samxx/post_timeloop.cpp

```diff
@@ -236,6 +247,11 @@ void post_timeloop() {
     vln  (k,icrm) = vln  (k,icrm) * factor_xy;
   });
 
+#if defined(MMF_VT)
+  // extra diagnostic step here for output tendencies
+  VT_diagnose();
+#endif
+
```

Comment:
@mrnorman Is the C++ code base adopting as a standard the use of preprocessor directives to implement optional behavior? If so, I worry this is a decision we may come to regret.

Reply:
I think we should shield the C++ code from these in `const.h`, though, by setting `constexpr bool` variables. @whannah1 , can you go into `samxx/samxx_const.h`, and put the following:
```
#ifdef MMF_VT
  bool constexpr use_mmf_vt = true;
#else
  bool constexpr use_mmf_vt = false;
#endif
```

Then use `if (use_mmf_vt) { ... }` instead of `ifdef`

Reply:
@pressel , note that there are circumstances when we cannot shield the code from `ifdef`s. The most notable of these situations is when the paramter list to a routine depends on the `ifdef`.

Reply:
Also, this might get hairy. Sometimes, you have to shield code with these, and the only ways to do that are `if constexpr`, which is not necessarily supported by all GPU compilers (notably nvcc the last time I checked); or to use `std::enable_if<>`, which is really ugly to have in user code. I'd honestly rather have `ifdef` than `enable_if` templates if it's asthetic we're talking about.

Reply:
@mrnorman I agree that ifdefs probably can't be avoided entirely, but I think should be avoid if possible without adding excess code complexity. We are on the same page.

Reply:
I've gone through and removed all ifdefs. I added 2 new namelist variables which are passed into the CRM that control whether VT is used and whether the bulk or filtered method is used.